From 304c1f39904d20578e676a9d9eb6cfe68db3f9b3 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
Date: Mon, 21 Jan 2019 11:01:00 +0100
Subject: [PATCH] DTS: Move "dio_spi" and "cpu0" into include file

Extend dio_spi by
 - GPIO definitions of the load and reset GPIO
 - CPOL (inverse polarity configuration)
---
 arch/arm/boot/dts/imx6q-vtpctp-0004.dts | 14 --------------
 arch/arm/boot/dts/imx6q-vtpctp-0005.dts | 14 --------------
 arch/arm/boot/dts/imx6q-vtpctp-0007.dts | 14 --------------
 arch/arm/boot/dts/imx6q-vtpctp-0010.dts | 14 --------------
 arch/arm/boot/dts/imx6qdl-vtpctp.dtsi   | 18 ++++++++++++++++++
 5 files changed, 18 insertions(+), 56 deletions(-)

diff --git a/arch/arm/boot/dts/imx6q-vtpctp-0004.dts b/arch/arm/boot/dts/imx6q-vtpctp-0004.dts
index 588bd94..8bfcd54 100644
--- a/arch/arm/boot/dts/imx6q-vtpctp-0004.dts
+++ b/arch/arm/boot/dts/imx6q-vtpctp-0004.dts
@@ -104,18 +104,4 @@
 
 		linux,wakeup;
 	};
-
-	di0_spi: dio_spi@1 {
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_dio_spi>;
-		compatible = "dio_spi";
-		reg = <1>;
-		spi-max-frequency = <1000000>;
-	};
-
-};
-
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1c_reg>;
 };
diff --git a/arch/arm/boot/dts/imx6q-vtpctp-0005.dts b/arch/arm/boot/dts/imx6q-vtpctp-0005.dts
index 15cca03..447e4af 100644
--- a/arch/arm/boot/dts/imx6q-vtpctp-0005.dts
+++ b/arch/arm/boot/dts/imx6q-vtpctp-0005.dts
@@ -95,18 +95,4 @@
 
 		linux,wakeup;
 	};
-
-	di0_spi: dio_spi@1 {
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_dio_spi>;
-		compatible = "dio_spi";
-		reg = <1>;
-		spi-max-frequency = <1000000>;
-	};
-
-};
-
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1c_reg>;
 };
diff --git a/arch/arm/boot/dts/imx6q-vtpctp-0007.dts b/arch/arm/boot/dts/imx6q-vtpctp-0007.dts
index 280b0c4..0a80b5e 100644
--- a/arch/arm/boot/dts/imx6q-vtpctp-0007.dts
+++ b/arch/arm/boot/dts/imx6q-vtpctp-0007.dts
@@ -98,18 +98,4 @@
 
 		linux,wakeup;
 	};
-
-	di0_spi: dio_spi@1 {
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_dio_spi>;
-		compatible = "dio_spi";
-		reg = <1>;
-		spi-max-frequency = <1000000>;
-	};
-
-};
-
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1c_reg>;
 };
diff --git a/arch/arm/boot/dts/imx6q-vtpctp-0010.dts b/arch/arm/boot/dts/imx6q-vtpctp-0010.dts
index 3b2cc4b..080570d 100644
--- a/arch/arm/boot/dts/imx6q-vtpctp-0010.dts
+++ b/arch/arm/boot/dts/imx6q-vtpctp-0010.dts
@@ -97,20 +97,6 @@
 
 		linux,wakeup;
 	};
-
-	di0_spi: dio_spi@1 {
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_dio_spi>;
-		compatible = "dio_spi";
-		reg = <1>;
-		spi-max-frequency = <1000000>;
-	};
-
-};
-
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1c_reg>;
 };
 
 &anatop {
diff --git a/arch/arm/boot/dts/imx6qdl-vtpctp.dtsi b/arch/arm/boot/dts/imx6qdl-vtpctp.dtsi
index e3a3333..c593b38 100644
--- a/arch/arm/boot/dts/imx6qdl-vtpctp.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-vtpctp.dtsi
@@ -1167,6 +1167,24 @@
 	regulator-allow-bypass;
 };
 
+&ecspi4 {
+	di0_spi: dio_spi@1 {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_dio_spi>;
+		compatible = "dio_spi";
+		reg = <1>;
+		spi-cpol;
+		spi-max-frequency = <1000000>;
+		gpio-load = <&gpio3 19 GPIO_ACTIVE_HIGH>;
+		gpio-reset = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&cpu0 {
+	arm-supply = <&sw1a_reg>;
+	soc-supply = <&sw1c_reg>;
+};
+
 /delete-node/ &ipu1_di0_hdmi;
 /delete-node/ &hdmi_mux_0;
 /delete-node/ &ipu1_di1_hdmi;
-- 
2.7.4

